rust/parquet/src/util/io.rs

```diff
@@ -243,6 +251,19 @@ mod tests {
         assert_eq!(src.pos(), 4);
     }
 
+    #[test]
+    fn test_io_slice_over_limit() {
```

Comment:
Is this testing the case when `buf.len() > src.len()` ?

Reply:
When you slice a reader in a way that it exceeds the bounds of the original reader, the slice reader should be clamped to the original reader. This is what is tested here.
```
Original:        ++++++++++++++++++++   
Requested Slice: ----------------+++++++++++++
Resulting Slice: ----------------++++
```